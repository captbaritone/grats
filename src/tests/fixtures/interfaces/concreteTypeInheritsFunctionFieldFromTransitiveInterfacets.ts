/** @gqlInterface */
interface Entity {
  name: string | null;
}

/** @gqlField */
export function greeting(entity: Entity): string {
  return `Hello, ${entity.name ?? "World"}!`;
}

/** @gqlInterface */
interface IThing extends Entity {
  name: string | null;
}

/** @gqlType */
export class Doohickey implements IThing, Entity {
  __typename: "Doohickey";
  name: string;
}
