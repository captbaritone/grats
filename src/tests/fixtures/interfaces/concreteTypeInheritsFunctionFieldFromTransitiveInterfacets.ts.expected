-----------------
INPUT
----------------- 
/** @gqlInterface */
interface Entity {
  name: string | null;
}

/** @gqlField */
export function greeting(entity: Entity): string {
  return `Hello, ${entity.name ?? "World"}!`;
}

/** @gqlInterface */
interface IThing extends Entity {
  name: string | null;
}

/** @gqlType */
export class Doohickey implements IThing, Entity {
  __typename: "Doohickey";
  name: string;
}

-----------------
OUTPUT
-----------------
-- SDL --
interface Entity {
  greeting: String @metadata(argCount: 1, name: "greeting", tsModulePath: "grats/src/tests/fixtures/interfaces/concreteTypeInheritsFunctionFieldFromTransitiveInterfacets.ts")
}

interface IThing implements Entity {
  greeting: String @metadata(argCount: 1, name: "greeting", tsModulePath: "grats/src/tests/fixtures/interfaces/concreteTypeInheritsFunctionFieldFromTransitiveInterfacets.ts")
}

type Doohickey implements Entity & IThing {
  greeting: String @metadata(argCount: 1, name: "greeting", tsModulePath: "grats/src/tests/fixtures/interfaces/concreteTypeInheritsFunctionFieldFromTransitiveInterfacets.ts")
}
-- TypeScript --
import { greeting as doohickeyGreetingResolver } from "./concreteTypeInheritsFunctionFieldFromTransitiveInterfacets";
import { GraphQLSchema, GraphQLInterfaceType, GraphQLString, GraphQLObjectType } from "graphql";
export function getSchema(): GraphQLSchema {
    const EntityType: GraphQLInterfaceType = new GraphQLInterfaceType({
        name: "Entity",
        fields() {
            return {
                greeting: {
                    name: "greeting",
                    type: GraphQLString
                }
            };
        }
    });
    const IThingType: GraphQLInterfaceType = new GraphQLInterfaceType({
        name: "IThing",
        fields() {
            return {
                greeting: {
                    name: "greeting",
                    type: GraphQLString
                }
            };
        },
        interfaces() {
            return [EntityType];
        }
    });
    const DoohickeyType: GraphQLObjectType = new GraphQLObjectType({
        name: "Doohickey",
        fields() {
            return {
                greeting: {
                    name: "greeting",
                    type: GraphQLString,
                    resolve(source) {
                        return doohickeyGreetingResolver(source);
                    }
                }
            };
        },
        interfaces() {
            return [EntityType, IThingType];
        }
    });
    return new GraphQLSchema({
        types: [EntityType, IThingType, DoohickeyType]
    });
}
